nrf52-hal-common/src/saadc.rs

```diff
@@ -96,12 +92,12 @@ impl Default for SaadcConfig {
     }
 }
 
-impl<PIN> OneShot<Saadc, u16, PIN> for Saadc
+impl<PIN> OneShot<Saadc, i16, PIN> for Saadc
```

Comment:
Why is this an i16? I am kinda sure counters can't do negative numbers? Or is the Nordic one kinda special? :)

Reply:
https://infocenter.nordicsemi.com/index.jsp?topic=%2Fcom.nordic.infocenter.sdk5.v11.0.0%2Fgroup__nrf__saadc__hal.html

The ADC is differential and I'm sure my testing showed it could go below zero. The existing code would give a number in the high 65k region.

Reply:
Oh cool, I didn't know that =)
I hope this is true for all the devices we support currently. But since it's in the generic HAL I think it's safe to assume so.
Maybe in the future we could encode the mode (differential, single ended) in a typestate.